lib/properties.js

```diff
@@ -267,22 +267,19 @@ internals.properties.prototype.parseString = function(property, joiObj) {
   const describe = joiObj.describe();
 
   if (describe.type !== 'date') {
-    property.minLength = internals.getArgByName(describe.rules, 'min');
-    property.maxLength = internals.getArgByName(describe.rules, 'max');
+    property.minLength = internals.getArgByName(describe.rules, 'min', 'limit');
```

Comment:
`getArgByName` now automatically returns:

1. The key specified if found
2. The first key if none specified (and options exist) --> Matches joi expected user experience consistently
3. `true` if no options exist.

This is the only place I'm specifying the key, mainly because it's a named top-level property within string. This maintains more compatibility than most keys.